### ISSUE TITLE
Please check why my config py file can't work for HTC.

### ISSUE THREAD

**Issue by minan19605** [open-mmlab/mmdetection#2826]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
I want to train an HTC model with my dataset which follows the coco dataset format. I spent about 3 days to prepare my config py files but can't work. The training procedure was always like below, the eta time is short and most loss values are 0.000 and the act value became 100.00 very quickly.
2020-05-26 16:54:34,497 - mmdet - INFO - workflow: [('train', 1)], max: 200 epochs
2020-05-26 16:55:00,487 - mmdet - INFO - Epoch [1][50/262]	lr: 0.00025, eta: 7:32:48, time: 0.519, data_time: 0.117, memory: 2802, loss_rpn_cls: 0.5122, loss_rpn_bbox: 0.0000, s0.loss_cls: 0.3147, s0.acc: 88.3418, s0.loss_bbox: 0.0000, s0.loss_mask: 0.0000, s1.loss_cls: 0.2343, s1.acc: 78.0215, s1.loss_bbox: 0.0000, s1.loss_mask: 0.0000, s2.loss_cls: 0.1145, s2.acc: 84.0449, s2.loss_bbox: 0.0000, s2.loss_mask: 0.0000, loss: 1.1757
2020-05-26 16:55:24,353 - mmdet - INFO - Epoch [1][100/262]	lr: 0.00050, eta: 7:14:12, time: 0.477, data_time: 0.088, memory: 2802, loss_rpn_cls: 0.0032, loss_rpn_bbox: 0.0000, s0.loss_cls: 0.2846, s0.acc: 100.0000, s0.loss_bbox: 0.0000, s0.loss_mask: 0.0000, s1.loss_cls: 0.1435, s1.acc: 100.0000, s1.loss_bbox: 0.0000, s1.loss_mask: 0.0000, s2.loss_cls: 0.0722, s2.acc: 100.0000, s2.loss_bbox: 0.0000, s2.loss_mask: 0.0000, loss: 0.5034

Finally, I deleted below sentences from bottom of my config file:
data = dict(
    train=dict(pipeline=train_pipeline),
    val=dict(pipeline=test_pipeline),
    test=dict(pipeline=test_pipeline))

The training procedure became normal.  Could you please check why this problem happened.

I attached two config files for your information.
1. the file: htc_r50_fpn_20e_coco_wheat_linux.py can't work.
2. the file: htc_without_semantic_r50_fpn_1x_coco_linux.py  is work. 
3. the log file has all details for the wrong training procedure.

And I have only one class, and I don't need the semantic function.

[htc_r50_fpn_20e_coco_wheat_linux.py.txt](https://github.com/open-mmlab/mmdetection/files/4686312/htc_r50_fpn_20e_coco_wheat_linux.py.txt)
[htc_without_semantic_r50_fpn_1x_coco_linux.py.txt](https://github.com/open-mmlab/mmdetection/files/4686314/htc_without_semantic_r50_fpn_1x_coco_linux.py.txt)

[20200526_165430.log.txt](https://github.com/open-mmlab/mmdetection/files/4686327/20200526_165430.log.txt)

**Comment by hellock** [comment]
What do your mean by `work` and `not work`? If you mean the code crashed, then it should be some issue with the code and you may follow the error template to provide necessary infomation, and we will help fix it. If you mean the performance is not good, then it is mostly about your own work like hyper-parameter tuning, data analysis, etc. You need to have a good understanding on your data and the method you adopted to make it work.

**Comment by minan19605** [comment]
Not work means the training procedure is abnormal.  The work means the training procedure is normal.
The abnormal means during the training procedure, the loss, and the accuracy value are abnormal. 
After I deleted the below sentence from my config py file, the training procedure became normal. So I want to know why the below sentence affects the training procedure.

data = dict(
train=dict(pipeline=train_pipeline),
val=dict(pipeline=test_pipeline),
test=dict(pipeline=test_pipeline))

**Comment by hellock** [comment]
Why you add the last several lines?

**Comment by minan19605** [comment]
Previously, I used the file "htc_r50_fpn_1x_coco.py" to create myself config py file. In the file "htc_r50_fpn_1x_coco.py", there have:
data = dict(
    train=dict(
        seg_prefix=data_root + 'stuffthingmaps/train2017/',
        pipeline=train_pipeline),
    val=dict(pipeline=test_pipeline),
    test=dict(pipeline=test_pipeline))
Because I don't have the stuff annotation files. so I deleted line "seg_prefix=data_root + 'stuffthingmaps/train2017/',"
Finally, I got the lines below which make training procedure abormal.
data = dict(
train=dict(pipeline=train_pipeline),
val=dict(pipeline=test_pipeline),
test=dict(pipeline=test_pipeline))

**Comment by hellock** [comment]
The config is messed up. You were not inheriting from base configs but overwriting the whole dict. Please refer to the [documentation](https://mmdetection.readthedocs.io/en/latest/config.html).
